src/math/trig.c

```diff
@@ -9,568 +9,97 @@
-	1817825448,
-	1821324571,
-	1824806551,
-	1828271355,
-	1831718951,
-	1835149305,
-	1838562387,
-	1841958164,
-	1845336603,
-	1848697673,
-	1852041343,
-	1855367580,
-	1858676354,
-	1861967633,
-	1865241387,
-	1868497585,
-	1871736195,
-	1874957188,
-	1878160534,
-	1881346201,
-	1884514160,
-	1887664382,
-	1890796836,
-	1893911493,
-	1897008324,
-	1900087300,
-	1903148391,
-	1906191569,
-	1909216806,
-	1912224072,
-	1915213339,
-	1918184580,
-	1921137766,
-	1924072870,
-	1926989863,
-	1929888719,
-	1932769410,
-	1935631909,
-	1938476189,
-	1941302224,
-	1944109986,
-	1946899450,
-	1949670588,
-	1952423376,
-	1955157787,
-	1957873795,
-	1960571374,
-	1963250500,
-	1965911147,
-	1968553291,
-	1971176905,
-	1973781966,
-	1976368449,
-	1978936330,
-	1981485584,
-	1984016188,
-	1986528117,
-	1989021349,
-	1991495859,
-	1993951624,
-	1996388621,
-	1998806828,
-	2001206221,
-	2003586778,
-	2005948477,
-	2008291295,
-	2010615209,
-	2012920200,
-	2015206244,
-	2017473320,
-	2019721407,
-	2021950483,
-	2024160528,
-	2026351521,
-	2028523441,
-	2030676268,
-	2032809981,
-	2034924561,
-	2037019987,
-	2039096240,
-	2041153301,
-	2043191149,
-	2045209766,
-	2047209132,
-	2049189230,
-	2051150040,
-	2053091543,
-	2055013722,
-	2056916559,
-	2058800035,
-	2060664132,
-	2062508835,
-	2064334123,
-	2066139982,
-	2067926393,
-	2069693341,
-	2071440807,
-	2073168776,
-	2074877232,
-	2076566159,
-	2078235539,
-	2079885359,
-	2081515602,
-	2083126253,
-	2084717297,
-	2086288719,
-	2087840504,
-	2089372637,
-	2090885104,
-	2092377891,
-	2093850984,
-	2095304369,
-	2096738031,
-	2098151959,
-	2099546138,
-	2100920555,
-	2102275198,
-	2103610053,
-	2104925108,
-	2106220351,
-	2107495769,
-	2108751351,
-	2109987084,
-	2111202958,
-	2112398959,
-	2113575079,
-	2114731304,
-	2115867625,
-	2116984030,
-	2118080510,
-	2119157053,
-	2120213650,
-	2121250291,
-	2122266966,
-	2123263665,
-	2124240379,
-	2125197099,
-	2126133816,
-	2127050521,
-	2127947205,
-	2128823861,
-	2129680479,
-	2130517051,
-	2131333571,
-	2132130029,
-	2132906419,
-	2133662733,
-	2134398965,
-	2135115106,
-	2135811152,
-	2136487094,
-	2137142926,
-	2137778643,
-	2138394239,
-	2138989707,
-	2139565042,
-	2140120239,
-	2140655292,
-	2141170196,
-	2141664947,
-	2142139540,
-	2142593970,
-	2143028233,
-	2143442325,
-	2143836243,
-	2144209981,
-	2144563538,
-	2144896909,
-	2145210091,
-	2145503082,
-	2145775879,
-	2146028479,
-	2146260880,
-	2146473079,
-	2146665075,
-	2146836865,
-	2146988449,
-	2147119824,
-	2147230990,
-	2147321945,
-	2147392689,
-	2147443221,
-	2147473541,
-	2147483647
-};
-
-/* Sine lookup table read */
-static inline int32_t sine_lookup(int idx)
+int32_t sin_fixed(int32_t th_rad_fxp)
 {
-	int32_t s;
-	int i1;
+	/*cordic_atan2_lookup_table = atan(2.^-(0:N-1)) N = 31 */
+	/*CORDIC Gain is cordic_gain = prod(sqrt(1 + 2.^(-2*(0:31-1)))) */
+	/*Inverse CORDIC Gain,inverse_cordic_gain = 1 / cordic_gain */
+	static const int32_t inpLUT[31] = { 843314857, 497837829, 263043837, 133525159,
```

Comment:
please give this a better name, inpLUT does not clearly state what it contains. A more appropriate name would be a truncated version fo the name you gave it above `atan2_lookup_table`

Reply:
OK, will replace it with a new name